core/src/test/java/io/ran/ClazzTest.java

```diff
@@ -250,4 +286,25 @@ public void setId(int id) {
 			this.id = id;
 		}
 	}
+
+	public static class GenericClass<T> {
+		public void method(T t) {
+
+		}
+	}
+
+	public static class GenericImpl extends GenericClass<String> {
+
+	}
+
+	public interface GenericInterface<T> {
+		void method(T t);
+	}
+
+	public static class GenericInterfaceImpl implements GenericInterface<String> {
```

Comment:
We're missing two examples:

```
public interface NonGenericInterface extends GenericInterface<String> {
    // This can be left empty, since method(T) is now method(String)
}
public interface NonGenericInterface2 extends NonGenericInterface {
    // This one is tricky because the method is declared deeper
}
public interface NonGenericInterfaceExplicit extends GenericInterface<String> {
    /**
     * Better java comment - more specific, and new parameter name
     */
    void method(String myString);
}

```
Also, my findings is that from the `NonGenericInterfaceExplicit` example I get two methods:
* method(Object)
* method(String)
Which is frustrating since I am only interested in the `method(String)` one...

Reply:
ad31ed3

The last one is actually correct, since there are two methods. I have added a helper method on the ClazzMethodList to make it easier to select a specific method

Reply:
see comment below.